app/src/main/java/com/appunite/loudius/ui/MainViewModel.kt

```diff
@@ -0,0 +1,35 @@
+package com.appunite.loudius.ui
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.appunite.loudius.network.utils.AuthFailureHandler
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+data class MainState(
+    val authFailureEvent: Unit? = null,
+)
+
+@HiltViewModel
+class MainViewModel @Inject constructor(private val authFailureHandler: AuthFailureHandler) :
+    ViewModel() {
+
+    var state by mutableStateOf(MainState())
+        private set
+
+    init {
+        viewModelScope.launch {
+            authFailureHandler.authFailureFlow.collect {
+                state = MainState(Unit)
+            }
+        }
+    }
+
+    fun onAuthFailureHandled() {
+        state = MainState(null)
```

Comment:
nitpick: null is the default value, so it could be
```suggestion
        state = MainState()
```